ext/standard/type.c

```diff
@@ -289,6 +289,43 @@ PHP_FUNCTION(is_array)
 }
 /* }}} */
 
+/* {{{ proto bool is_list(mixed var)
+   Returns true if variable is an array whose keys are all numeric, sequential,
+   and start at 0 */
+PHP_FUNCTION(is_list)
+{
+	zval *arg;
+	zend_array *arrval;
+	zend_ulong num_idx, expected_idx = 0;
+	zend_string *str_idx;
+
+	ZEND_PARSE_PARAMETERS_START(1, 1)
+		Z_PARAM_ZVAL(arg)
```

Comment:
I think you could use `Z_PARAM_ARRAY_HT()` here, to get `arrval` without further ado.

Reply:
That would make the function throw an error when passed a non-array argument, which isn't the intended behavior:

```
Fatal error: Uncaught TypeError: is_list() expects parameter 1 to be array, null given
```

Reply:
@duskwuff AFAIK, you could also use the following if you want to achieve an `is_*` alike behaviour:
```C
ZEND_PARSE_PARAMS_START_EX(ZEND_PARSE_PARAMS_QUIET, 1, 1)
// ...
```
(Although I would also prefer constraining the type to `arrays`)

Reply:
@duskwuff, ah, I see. Thanks.